rs/http_endpoints/public/src/lib.rs

```diff
@@ -337,23 +338,27 @@ pub fn start_server(
     .build();
 
     let call_router = call::CallServiceV2::new_router(call_handler.clone());
-    let (ingress_watcher_handle, _) = IngressWatcher::start(
-        rt_handle.clone(),
-        log.clone(),
-        metrics.clone(),
-        certified_height_watcher,
-        completed_execution_messages_rx,
-        CancellationToken::new(),
-    );
 
-    let call_v3_router = call::CallServiceV3::new_router(
-        call_handler,
-        ingress_watcher_handle,
-        metrics.clone(),
-        config.ingress_message_certificate_timeout_seconds,
-        delegation_from_nns.clone(),
-        state_reader.clone(),
-    );
+    let call_v3_router = if enable_v3_call_handler {
+        let (ingress_watcher_handle, _) = IngressWatcher::start(
+            rt_handle.clone(),
+            log.clone(),
+            metrics.clone(),
+            certified_height_watcher,
+            completed_execution_messages_rx,
+            CancellationToken::new(),
+        );
+        Some(call::CallServiceV3::new_router(
+            call_handler,
+            ingress_watcher_handle,
+            metrics.clone(),
+            config.ingress_message_certificate_timeout_seconds,
+            delegation_from_nns.clone(),
+            state_reader.clone(),
+        ))
+    } else {
+        None
+    };
```

Comment:
How does the axum routing work if one route is `api/:api_version/canister/:effective_canister_id/call"` and `api/v3/canister/:effective_canister_id/call` also exists?